project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module

```diff
@@ -93,6 +82,58 @@ function c4m_features_homepage_c4m_suggested_groups_get_groups() {
 }
 
 /**
+ * Renders 3 featured groups.
+ *
+ * @return mixed
+ *   Rendered HTML of 3 featured groups.
+ */
+function c4m_features_homepage_c4m_featured_groups_get_groups() {
+  $query = new EntityFieldQuery();
+
+  $query
+    ->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'group')
+    ->propertyCondition('status', NODE_PUBLISHED)
+    ->propertyCondition('promote', NODE_PROMOTED);
+
+  $result = $query
+    ->addTag('random')
+    ->range(0, 3)
+    ->execute();
+
+  if (empty($result['node'])) {
+    return;
+  }
+
+  $groups = c4m_features_homepage_alphabetically_render_node_list($result['node']);
+
+  $variables = array(
+    'groups' => $groups,
+  );
+
+  return theme('c4m_features_homepage_c4m_featured_groups', $variables);
+}
+
+/**
+ * Loads list of nodes, sorts them and renders them as block list.
+ */
+function c4m_features_homepage_alphabetically_render_node_list($node_ids) {
+  $items = array();
+  foreach ($node_ids as $node) {
+    $items[] = node_load($node->nid);
```

Comment:
@DavidHernandez 
Can we use `node_load_multiple` instead ?
(better to hit the DB only once)